src/common/slices/iconSlice.ts

```diff
@@ -0,0 +1,121 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { getServiceClient, URLS } from '../services';
+import { typeIconInfos } from './icons';
+
+export interface IconInfo {
+  icon?: string;
+  color: string;
+  url?: string;
+  isImage: boolean;
+}
+interface AppIconCache {
+  release: { [key: string]: IconInfo };
+  beta: { [key: string]: IconInfo };
+  dev: { [key: string]: IconInfo };
+}
+
+export enum AppTag {
+  release = 'release',
+  beta = 'beta',
+  dev = 'dev',
+}
+
+interface IconState {
+  typeIconInfos: { [key: string]: IconInfo };
+  appIconCache: AppIconCache;
+  defaultApp: IconInfo;
+  defaultType: IconInfo;
+  loadingIcon: IconInfo;
+}
+
+type AppIconPayload = {
+  appId: string;
+  appTag: AppTag;
+  icon?: IconInfo;
```

Comment:
Probably being pedantic but... in what situation is this undefined? if it's that one is the input for the thunk, and one is the output, consider using two different types